CONTRIBUTING.md

```diff
@@ -0,0 +1,194 @@
+# Developer Guidelines
+
+## Code Style
+
+### Existing non-compliant code
+The code that does not comply with these guidelines will be fixed in separate pull request(s). Any new contribution should follow closely these guidelines but should not change style in the existing GridKit code.
+
+### Error handling
+Return values of member functions should be of type `int` and used for error handling. Functions return 0 if no error is encounter, return positive value for warnings and recoverable error, and negative value for irrecoverable errors.
+
+### Output
+If an output is needed (for example, a warning needs to be displayed), use `std::cout` and not `printf` as shown below. There should be a space before and after each `<<`.
+
+```c++
+std::cout << "index out of bounds. Row " << i << " starts at: " << start << " and ends at " << end << std::endl;
+```
+
+
+### Member variable naming
+
+Member variable names should use C-style name format and end with trailing underscore `_`.
+```c++
+double member_variable_; // Yes
+double another_member;   // No, there is no trailing underscore to distinguish it from nonmember variables
+double memberVariable_;  // No, using lowercase camel instead of C-style name format
+```
+
+### Function names
+
+Use lowercase camel format for function names.
+```c++
+int myFunction(double x); // Yes
+int another_function();   // No, using C-style name format
+int YetAnotherFunction(); // No, using uppercase camel name format
+```
+
+### Class names
+
+Class names should start with a capital letter. For instance, `Vector` and `Matrix` are valid class names, while `point` is not.
+
+### Enums (enumerated types)
+
+Always define `enum`s inside `GridKit` namespace. Type names should be capitalized and the constant names should be uppercase with underscores (but there is no underscore at the end!).
+
+```c++
+  enum ExampleEnum { CONST_ONE = 0,
+                     CONST_TWO = 8, 
+                     YET_ANOTHER_CONST = 17 };
+``` 
+
+### Constants
+
+If a constant is used in more than one file, define it in `Common.h`. Constants names should be capitalized.
+
+```c++
+   constexpr double Pi = 3.1415; // No, it should be all caps
+   constexpr double SQRT_TWO = 1.4142 // No, there is an underscore
+   constexpr double SQRTTWO_ = 1.4142 // No, there is an underscore
+   constexpr double EXP = 2.7183 // Yes   
+```
+
+### Pointers and references
+
+The pointer `*` or reference `&` belong to the type and there should be no space between them and the type name.
+```c++
+double* x;     // Yes
+int& n;        // Yes
+double *x, *y; // No, the pointer symbol is a part of `double*` type
+int & n;       // No, the reference symbol is a part of `int&` type
+```
+
+### Indentation
+Use only spaces for indentation, not tabs. Indent size is 4 spaces.
```

Comment:
I'm fine with this if this really is the preference of everyone else, however this could be considered an accessibility issue. With tabs for indentation, people who are hard of sight can increase the indentation size to their preference while others can keep it as low as they want.

What I typically suggest is a tabs for indentation, spaces for alignment policy for this reason. This should be easy to enact with a `.clang-format` file.